### DISCUSSION TITLE
Can Dagster Cloud fetch compute logs from another service like Grafana rather than storing it on the Dagster Cloud servers?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18064] [Q&A] [answered]
Dagster Cloud (Hybrid) + Compute Logs question- The <https://docs.dagster.io/dagster-cloud/deployment/hybrid#runs|docs> say that run workers upload all compute logs to Dagster Cloud. This seems like it could be too much data to store - and for privacy reasons, ideally Dagster wouldn’t store all our logs. We currently ingest all stdout/stderr logs into Loki from processes kicked-off by our existing scheduler - so ideally Dagster Cloud’s UI would ask the code deployment to query Loki for logs when it needs to display them (or at least grab them from our own S3 bucket or similar.) How are others dealing with this? Does Dagster Cloud just host its own copy of all your process logs - and you keep another copy of logs in CloudWatch/Loki/Splunk for search, alerting, etc.?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C02LJ7G0LAZ/p1700121216696229?thread_ts=1700121216.696229&cid=C02LJ7G0LAZ).*

**Comment by prha** [comment, marked as answer]
For stdout/stderr, you can disable the upload of compute logs to Dagster Cloud by configuring the `NoOpComputeLogManager`: https://docs.dagster.io/dagster-cloud/deployment/agents/customizing-configuration#disabling-compute-logs

Separately, you could configure a cloud-based compute log manager to capture compute logs to be uploaded to your private cloud storage (e.g. private S3 bucket), but you would lose the UI integration where the compute logs can be visible in the run view.  You could still look up the logs, which would be arranged by run id.  For the S3 compute log manager specifically, there is a configuration where you can upload logs to your private S3 bucket, and the URL of your private bucket console page is displayed in the run view (for easier navigation).  See the `show_url_only` option in https://docs.dagster.io/_apidocs/libraries/dagster-aws#dagster_aws.s3.S3ComputeLogManager

The compute log manager is designed to be pluggable.  You could implement your own integration with Grafana to display a link to a particular run worker's logs.
